libs/service-portal/health/src/screens/OrganDonationRegistration/RegistrationForm.tsx

```diff
@@ -20,24 +21,39 @@ import * as styles from './OrganDonationRegistration.css'
 import Limitations from './Limitations'
 import { useNavigate } from 'react-router-dom'
 import {
-  useGetDonorStatusQuery,
-  useGetOrganDonationExceptionsQuery,
+  useGetOrgansListQuery,
   useUpdateOrganDonationInfoMutation,
 } from '../OrganDonation/OrganDonation.generated'
+import { Loader } from './Loader'
+
+const OPT_IN = 'opt-in'
+const OPT_IN_EXCEPTIONS = 'opt-in-exceptions'
+const OPT_OUT = 'opt-out'
 
 export const Form2 = () => {
   useNamespaces('sp.health')
   const { formatMessage, lang } = useLocale()
   const navigate = useNavigate()
 
-  const OPT_IN = 'opt-in'
-  const OPT_IN_EXCEPTIONS = 'opt-in-exceptions'
-  const OPT_OUT = 'opt-out'
-
-  const { data, loading } = useGetOrganDonationExceptionsQuery({
+  const { data, loading } = useGetOrgansListQuery({
     variables: { locale: lang },
   })
 
+  const isDonor = data?.HealthDirectorateOrganDonation.donor?.isDonor
+  const hasLimitations =
+    data?.HealthDirectorateOrganDonation.donor?.limitations?.hasLimitations
+  const allLimitations = data?.HealthDirectorateOrganDonation.organList
+  const selectedLimitations =
+    data?.HealthDirectorateOrganDonation.donor?.limitations?.limitedOrgansList?.map(
+      (item) => item.id,
+    )
+  const donorStatus = isDonor
+    ? hasLimitations
+      ? OPT_IN_EXCEPTIONS
+      : OPT_IN
+    : OPT_OUT
```

Comment:
I think this would be better in the service, returning a enum `donorStatus `with the strings defined above. IMO its cleaner to keep the business logic out back.

Reply:
Im also using it to keep track of users choice